server/app/services/grid_service_deployer.rb

```diff
@@ -39,53 +39,67 @@ def selected_nodes
     nodes
   end
 
+  # @return [Boolean]
   def deploy
-    info "starting to deploy #{self.grid_service.to_path}"
-    self.grid_service_deploy.set(:deploy_state => :ongoing)
     deploy_rev = Time.now.utc.to_s
+    info "starting to deploy #{self.grid_service.to_path} at #{deploy_rev}"
+    self.grid_service_deploy.set(:deploy_state => :ongoing)
     self.grid_service.set(:deployed_at => deploy_rev)
 
-    deploy_futures = []
     total_instances = self.instance_count
     self.grid_service.grid_service_instances.where(:instance_number.gt => total_instances).destroy
```

Comment:
Hmm... the scheduler still uses the `Container` info for scheduling decisions, not the `ServiceInstance` models? Would need to fix that if you want the scheduler to re-schedule the new instances based on the old instances being gone, plus wait for the containers to terminate and release their resources.